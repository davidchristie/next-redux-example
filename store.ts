import { Action, applyMiddleware, createStore, Middleware } from "redux";
import createSagaMiddleware, { Task } from "redux-saga";
import rootReducer, { exampleInitialState, RootState } from "./reducer";
import { rootSaga } from "./sagas";

const bindMiddleware = (middleware: Middleware[]) => {
  if (process.env.NODE_ENV !== "production") {
    const { composeWithDevTools } = require("redux-devtools-extension");
    return composeWithDevTools(applyMiddleware(...middleware));
  }
  return applyMiddleware(...middleware);
};

export const createReduxStore = (initialState: Partial<RootState> = {}) => {
  const preloadedState = {
    ...exampleInitialState,
    ...initialState
  };
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore<
    RootState,
    Action<unknown>,
    { sagaTask: Task },
    unknown
  >(rootReducer, preloadedState, bindMiddleware([sagaMiddleware]));
  store.sagaTask = sagaMiddleware.run(rootSaga);
  return store;
};
