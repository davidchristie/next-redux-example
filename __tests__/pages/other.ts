import { NextPageContext } from "next";
import OtherPage from "../../pages/other";
import {
  describeComponent,
  itShouldMatchSnapshot,
  itShowsText
} from "../../test-utils";
import { createReduxStore } from "../../store";

describeComponent(OtherPage, {}, getContext => {
  itShowsText(getContext, "Other Page");

  itShouldMatchSnapshot(getContext);

  describe("Initial Props", () => {
    it("Should match snapshot", async () => {
      const initialProps = await OtherPage.getInitialProps!({
        AppTree: () => null,
        ctx: {
          store: createReduxStore(),
          isServer: false
        },
        pathname: "/",
        query: {}
      } as NextPageContext);
      expect(initialProps).toMatchSnapshot();
    });
  });
});
