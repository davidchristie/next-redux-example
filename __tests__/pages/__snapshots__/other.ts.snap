// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherPage Component Should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Other Page
    </h1>
    <div
      class="jsx-1453841751 "
    >
      00:00:00
    </div>
    <div
      class="jsx-1118913787"
    >
      <h1
        class="jsx-1118913787"
      >
        Count: 
        <span
          class="jsx-1118913787"
        >
          0
        </span>
      </h1>
      <button
        class="jsx-1118913787"
      >
        +1
      </button>
      <button
        class="jsx-1118913787"
      >
        -1
      </button>
      <button
        class="jsx-1118913787"
      >
        Reset
      </button>
    </div>
    <nav>
      <a
        href="/"
      >
        Navigate: Index Page
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
